commit bde3440958114694a20dae6a2e95f1f99e917c2e
Author: David Rheinsberg <david@readahead.eu>
Date:   Fri Feb 20 10:18:38 2026 +0100

    util/sockopt: pass socket addresses as `struct sockaddr`
    
    Ensure that the sockopt-test passes addresses as `struct sockaddr *`
    rather than the typed `struct sockaddr_un*`, to honor the prototypes of
    the respective socket syscalls.
    
    Reported-by: Kostadin Shishmanov <kostadinshishmanov@protonmail.com>
    Signed-off-by: David Rheinsberg <david@readahead.eu>

diff --git a/src/util/test-sockopt.c b/src/util/test-sockopt.c
index eaa8fba..d859e08 100644
--- a/src/util/test-sockopt.c
+++ b/src/util/test-sockopt.c
@@ -28,7 +28,11 @@ static void create_server(int fd) {
         struct sockaddr_un address = { .sun_family = AF_UNIX };
         int r;
 
-        r = bind(fd, &address, offsetof(struct sockaddr_un, sun_path));
+        r = bind(
+                fd,
+                (const struct sockaddr *)&address,
+                offsetof(struct sockaddr_un, sun_path)
+        );
         c_assert(r >= 0);
 
         r = listen(fd, 256);
@@ -46,7 +50,11 @@ static void create_client(
         fd = socket(AF_UNIX, SOCK_STREAM | SOCK_NONBLOCK | SOCK_CLOEXEC, 0);
         c_assert(fd >= 0);
 
-        r = connect(fd, address, n_address);
+        r = connect(
+                fd,
+                (const struct sockaddr *)address,
+                n_address
+        );
         c_assert(r >= 0);
 
         *fdp = fd;
@@ -169,7 +177,11 @@ static void test_peerpidfd(void) {
         c_assert(l == 1 && c == '!');
 
         n_address = sizeof(address);
-        r = getsockname(fd_server, &address, &n_address);
+        r = getsockname(
+                fd_server,
+                (struct sockaddr *)&address,
+                &n_address
+        );
         c_assert(r >= 0);
 
         /*
